server/server.cpp

```diff
@@ -12446,6 +12465,24 @@ int main() {
             shutdownMode = SettingsManager::getIntSetting( "shutdownMode", 0 );
             forceShutdownMode = 
                 SettingsManager::getIntSetting( "forceShutdownMode", 0 );
+
+            int logLevel = SettingsManager::getIntSetting( "logLevel", 4 );
+            
+            switch(logLevel) {
+                case 4:
+                    logLevel = Log::INFO_LEVEL;
+                    break;
+                case 5:
+                    logLevel = Log::DETAIL_LEVEL;
+                    break;
+                case 6:
+                    logLevel = Log::TRACE_LEVEL;
+                    break;
+                default:
+                    logLevel = Log::INFO_LEVEL;
+                }
+
+            AppLog::setLoggingLevel( logLevel );
```

Comment:
What is the reason this is duplicated? Is it not possible to access the setting from the main function? (Line 12058)

Reply:
the part in the main function runs when server starts, this part runs periodically, so the setting takes effect without server restart.